lib/lotus/application_name.rb

```diff
@@ -0,0 +1,95 @@
+module Lotus
+  # An application name.
+  #
+  # @since x.x.x
+  class ApplicationName
+
+    # A list of words that are prohibited from forming the application name
+    #
+    # @since x.x.x
+    INVALID_WORDS = %w(lotus).freeze
```

Comment:
Maybe `RESERVED_WORDS` is more appropriate. If you check the next code comment you'll find yourself using the same term:

> Initialize and check against reserved words